In [34]:
import numpy as np
import pandas as pd
%matplotlib inline
import matplotlib.pyplot as plt
import PIL.Image as Image

### load the npz file and create the array to be then used for model training and testing

In [35]:
with np.load('images_data.npz', allow_pickle=True) as npz_file:
    df = pd.DataFrame(npz_file['values'], columns= npz_file['columns'])
    
df_train = df[df['imageSet'] == 'train']
X_tr = df_train.drop(['imageSet','category','imageFileName'], axis=1).values
y_tr = df_train.category.values

df_test = df[df['imageSet'] == 'test']
X_te = df_test.drop(['imageSet','category','imageFileName'], axis=1).values
y_te = df_test.category.values

features_names=df_train.drop(['imageSet','category','imageFileName'], axis=1).columns
categories = ['bike', 'car', 'motorcycle','other','truck','van']

### let's create a RandomForset Classifier:
- let start with 200 estimators and a depth of 6

In [36]:
from sklearn.ensemble import RandomForestClassifier

# Create a decision tree
rf = RandomForestClassifier(n_estimators=200, max_depth=6, random_state=0)

# Fit estimator
rf.fit(X_tr, y_tr)

RandomForestClassifier(bootstrap=True, class_weight=None, criterion='gini',
            max_depth=6, max_features='auto', max_leaf_nodes=None,
            min_impurity_decrease=0.0, min_impurity_split=None,
            min_samples_leaf=1, min_samples_split=2,
            min_weight_fraction_leaf=0.0, n_estimators=200, n_jobs=None,
            oob_score=False, random_state=0, verbose=0, warm_start=False)

In [37]:
rf.score(X_te, y_te)

0.96

- we reach an accuracy of 96% with RandomForest which is equivalent to k-NN
- the best accuracy with test set is obtained with a depth of 6

### let's create SVMs:
- linear
- RBF
- compare the results

In [38]:
from sklearn.svm import LinearSVC

linear_svc = LinearSVC(random_state=0, C=1, tol=0.0001,  max_iter=1000)
linear_svc.fit(X_tr, y_tr)
linear_svc.score(X_te, y_te)

0.9

- The linear SVM provides an accuracy of 0.9

In [40]:
from sklearn.svm import SVC

svc_rbf = SVC(kernel='rbf')
svc_rbf.fit(X_tr, y_tr)
svc_rbf.score(X_te, y_te)

C:\Users\lmbhoc2\AppData\Local\Continuum\anaconda3\envs\exts-ml\lib\site-packages\sklearn\svm\base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


0.94

- the SVM linear get a 90% score
- the SVM with RBF peforms good with 0.94 accuracy